### PR TITLE
Twitter: Move config to key value

### PR DESCRIPTION
**Pull request by annando**
Belongs to https://github.com/friendica/friendica/pull/12718

### CODE REVIEW

File: twitter/twitter.php [hunk 1]
```diff
@@ -1288,7 +1288,7 @@ function twitter_fetchtimeline(int $uid): void
 	$osecret = DI::pConfig()->get($uid, 'twitter', 'oauthsecret');
 	$lastid  = DI::pConfig()->get($uid, 'twitter', 'lastid');
 
-	$application_name = DI::config()->get('twitter', 'application_name');
+	$application_name = DI::keyValue()->get('twitter_application_name');
```
Comment (MrPetovan) [inline twitter/twitter.php]:
```suggestion
	$application_name = DI::keyValue()->get('twitter_application_name') ?? '';
```

File: twitter/twitter.php [hunk 2]
```diff
@@ -2187,7 +2187,7 @@ function twitter_fetchhometimeline(int $uid): void
 
 	Logger::info('Fetching timeline', ['uid' => $uid]);
 
-	$application_name = DI::config()->get('twitter', 'application_name');
+	$application_name = DI::keyValue()->get('twitter_application_name');
```
Comment (MrPetovan) [inline twitter/twitter.php]:
```suggestion
	$application_name = DI::keyValue()->get('twitter_application_name') ?? '';
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: twitter/twitter.php (hunk 1 above)
Comment (annando) [reply twitter/twitter.php]: I think that the behaviour of these two classes is identical for a non found value.

Comment (MrPetovan) [reply twitter/twitter.php]: Yes, it returns `null`, which is incorrect since we use the return value as a string.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**